app/src/main/java/ch/epfl/qedit/view/LoginActivity.java

```diff
@@ -1,41 +1,133 @@
 package ch.epfl.qedit.view;
 
+import android.content.Context;
 import android.content.Intent;
+import android.content.res.Resources;
 import android.os.Bundle;
 import android.os.Handler;
 import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Button;
 import android.widget.EditText;
 import android.widget.ProgressBar;
+import android.widget.Spinner;
 import android.widget.Toast;
 import androidx.appcompat.app.AppCompatActivity;
 import ch.epfl.qedit.R;
 import ch.epfl.qedit.backend.auth.AuthenticationFactory;
 import ch.epfl.qedit.backend.auth.AuthenticationService;
 import ch.epfl.qedit.model.User;
 import ch.epfl.qedit.util.Callback;
+import ch.epfl.qedit.util.LocaleHelper;
 import ch.epfl.qedit.util.Response;
 import ch.epfl.qedit.view.home.HomeActivity;
+import java.util.Locale;
 
-public class LoginActivity extends AppCompatActivity {
+public class LoginActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {
 
     public static final String USER = "ch.epfl.qedit.view.USER";
 
     private EditText tokenText;
+    private Button loginButton;
     private ProgressBar progressBar;
 
     private AuthenticationService authService;
     private Handler handler;
 
+    private boolean userHasInteracted = false;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
 
         tokenText = findViewById(R.id.login_token);
+        loginButton = findViewById(R.id.login_button);
         progressBar = findViewById(R.id.login_progress_bar);
 
         authService = AuthenticationFactory.getInstance();
         handler = new Handler();
+
+        /* Language selection */
+        // Create spinner (language list)
+        Spinner languageSelectionSpinner = findViewById(R.id.language_selection);
+
+        // Find app's current language position in languages list
+        String currentLanguage = Locale.getDefault().getLanguage();
+        int positionInLanguageList = 0;
+        String[] languageList = getResources().getStringArray(R.array.languages_codes);
+        boolean languageFound = false;
```

Comment:
We should probably use `Arrays.asList(languageList).indexOf(currentLanguage)` for more conciseness ;) ...

Reply:
That's very clean, I will do this, thank you!